x-pack/metricbeat/module/meraki/appliance_uplink_overview/appliance_uplink_overview.go

```diff
@@ -0,0 +1,102 @@
+package appliance_uplink_overview
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/v7/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/v7/metricbeat/mb"
+	"github.com/elastic/beats/v7/x-pack/metricbeat/module/meraki"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+
+	meraki_api "github.com/meraki/dashboard-api-go/v3/sdk"
+)
+
+// init registers the MetricSet with the central registry as soon as the program
+// starts. The New function will be called later to instantiate an instance of
+// the MetricSet for each host is defined in the module's configuration. After the
+// MetricSet has been created then Fetch will begin to be called periodically.
+func init() {
+	mb.Registry.MustAddMetricSet(meraki.ModuleName, "appliance_uplink_overview", New)
+}
+
+type config struct {
+	BaseURL       string   `config:"apiBaseURL"`
+	ApiKey        string   `config:"apiKey"`
+	DebugMode     string   `config:"apiDebugMode"`
+	Organizations []string `config:"organizations"`
+	// todo: device filtering?
```

Comment:
has there been any requirements come wrt filtering based on device attributes?

Reply:
Non that I am aware at this point.